src/main/java/com/scalyr/util/Util.java

```diff
@@ -0,0 +1,52 @@
+package com.scalyr.util;
+
+import com.scalyr.api.logs.EventAttributes;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public final class Util {
+  public static Integer stringToIntMemory(String maxBufferRam) {
+    if(maxBufferRam != null && !"".equals(maxBufferRam)) {
+      maxBufferRam = maxBufferRam.toLowerCase().trim();
+      if(maxBufferRam.contains("m")) {
+        return Integer.valueOf(maxBufferRam.substring(0, maxBufferRam.indexOf("m"))) * 1048576;
+      } else if (maxBufferRam.contains("k")) {
+        return Integer.valueOf(maxBufferRam.substring(0, maxBufferRam.indexOf("k"))) * 1024;
+      } else {
+        return Integer.valueOf(maxBufferRam);
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Mutation feels dirty...
+   * @param serverAttributes going to be mutated.
+   * @param extraAttributes kv string
+   */
+  public static void addExtraAttributesToServerAttributes(EventAttributes serverAttributes, String extraAttributes) {
+    for (Map.Entry<String, String> e: Util.kvStringToMap(extraAttributes).entrySet()) {
+      serverAttributes.put(e.getKey(), e.getValue());
+    }
+  }
+
+  public static Map<String, String> kvStringToMap(String str) {
+    return kvStringToMap(str, ",", "=");
+  }
+
+  public static Map<String, String> kvStringToMap(String str, String pairSeparator, String kvSeparator) {
+    Map<String, String> kvMap = new HashMap<String, String>();
+    String[] pairs = str.split(pairSeparator);
```

Comment:
I'd suggest:
* Check for str==null or empty, and return an empty map.
* Extra credit: call String.trim() on kv[0] and kv[1] before inserting into the map, to clean up extraneous whitespace. (It's easy for users to add whitespace by mistake, especially after a comma.)

Reply:
fixed these and added some unit tests around the Util class.